packages/amazon-sumerian-hosts-core/src/core/awspack/LexFeature.js

```diff
@@ -0,0 +1,287 @@
+import LexUtils from 'core/awspack/LexUtils';
+
+/**
+ * The AWS LexRuntime service object.
+ * @external LexRuntime
+ * @see https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/LexRuntime.html
+ */
+
+/**
+ * Feature class for interacting with Lex bot which the response from lex bot can be used for speech or other purpose.
+ *
+ * @property {Object} LEX_DEFAULTS - Default values to use with calls to {@link external:LexRuntime}.
+ * @property {string} [LEX_DEFAULTS.SampleRate='16000']
+ * @property {Object} EVENTS - Built-in messages that the feature emits.
+ * @property {string} [EVENTS.lexResponse=onLexResponseEvent] - Message that is emitted after
+ * receiving lex response for the input sent
+ * @property {string} [EVENTS.micReady=onMicrophoneReadyEvent] - Message that is emitted after
+ * microphone is ready to use
+ * @property {string} [EVENTS.micReady=onMicrophoneBeginRecordingEvent] - Message that is emitted after
+ * microphone starts recording
+ * @property {string} [EVENTS.micReady=onMicrophoneEndRecordingEvent] - Message that is emitted after
+ * microphone ends recording
+ * @property {Object} SERVICES - AWS services that are necessary for the feature
+ * to function.
+ * @property {external:LexRuntime} SERVICES.lexRuntime - The LexRuntime service that is used
+ * to interact with lex bot. Will be undefined until [initializeService]{@link LexFeature.initializeService}
+ * has been successfully executed
+ */
+class LexFeature{
+  /**
+   * @constructor
+   *
+   * @param {Messenger=} messenger - Optional Messenger object which the event will be emitted through
+   * @param {Object=} options - Options that will be used to interact with lex bot.
+   * @param {string=} options.botName - The name of the lex bot.
+   * @param {string=} options.botAlias - The alias of the lex bot.
+   * @param {string=} options.userId - The userId used to keep track of the session with lex bot. 
+  */
+  constructor(
+    messenger = undefined,
+    options = {
+      botName: undefined,
+      botAlias: undefined,
+      userId: undefined,
+    }
+  ) {
+    this._messenger = messenger;
+
+    this._botName = options.botName;
+    this._botAlias = options.botAlias;
+    this._userId = options.userId ? options.userId : Utils.createId();
+
+    //Microphone related fields
+    this._recording = false;
+    this._recLength = 0;
+    this._recBuffer = [];
+    this._setAudioContext();
+  }
+
+  /**
+   * Store LexRuntime and AWS SDK Version for use across all instances.
+   *
+   * @param {external:LexRuntime} lexRuntime - lexRuntime instance to use to interact with lex bot.
+   */
+  static initializeService(lexRuntime) {
+    // Make sure all were defined
+    if (lexRuntime === undefined)
+    {
+      throw new Error(
+        'Cannot initialize Lex feature. LexRuntime must be defined'
+      );
+    }
+
+    // Add sumerian hosts user-agent
+    if (lexRuntime.config) {
+      lexRuntime.config.customUserAgent = Utils.withCustomUserAgent(
```

Comment:
we should definitely use a customUserAgent so we know it is coming from Sumerian Hosts, but do we need an additional level of customization? is there a reason customers would want to add to this?

Reply:
This is to append sumerian hosts as an additional user-agent to whatever existing user agent a customer might already configured with his aws config

Reply:
oh got it, because the customer gives us the SDK client, we don't want to overwrite their agent. that makes sense

Reply:
> In addition to Preston's comment, as a customer, I would expect that if I explicitly provide a custom user agent string then _that_ is the exact string that will be used, not some modified version that concatenates my string with something else (which is what I believe this code does?)

I will keep with the current design for now since i don't think appending our agent to whatever existing agent affecting customers to trace through their own againt